cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -127,22 +325,106 @@ private static PropertyBag createPowerFlowProperties(CgmesModel cgmes, Terminal
         return p;
     }
 
+    // added TopologicalIsland as it was in cgmes : original topology is
+    // preserved.
+    private void addTopologicalIslandsToCgmes() {
+        if (!originalTopologicalIslands.isEmpty()) {
+            // For properties such as "cim:TopologicalIsland.TopologicalNodes", which might
+            // have arbitrary number of values, SPARQL will return multiple result sets.
+            // All properties values will be equal, except the multiValued property
+            // TopologicalNodes.
+            // Also, there can be > 1 TopologicalIsland, so we need to re-group PropertyBags
+            // by TopologicalIsland ID.
+            Map<String, PropertyBags> byTopologicalIslandId = new HashMap<>();
+            originalTopologicalIslands.forEach(pb -> {
+                String island = pb.getId(CgmesNames.TOPOLOGICAL_ISLAND);
+                if (byTopologicalIslandId.keySet().contains(island)) {
+                    byTopologicalIslandId.get(island).add(pb);
+                } else {
+                    PropertyBags pbs = new PropertyBags();
+                    pbs.add(pb);
+                    byTopologicalIslandId.put(island, pbs);
+                }
+            });
+            // now we can process all TPNodes from each island, and put them in one
+            // multivaluedProperty.
+            PropertyBags topologicalIslands = new PropertyBags();
+            byTopologicalIslandId.values().forEach(island -> {
+                PropertyBag topologicalIsland = new PropertyBag(SV_TOPOLOGICALISLAND_PROPERTIES);
+                topologicalIsland.setClassPropertyNames(Arrays.asList(CgmesNames.NAME));
```

Comment:
You can replace `Arrays.asList(CgmesNames.NAME)` by `Collections.singletonList(...)`

Reply:
Done